Remove inline js for gantt chart

bbovenzi opened the pull request:
Continuation of #14115.
- Removing inline js for the Gantt chart. 
- Linting all Gantt js
- Use more conventional file name


---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

bbovenzi committed:
lint gantt chart js and remove inline js

ryanahamilton reviewed (approved):
Nice work @bbovenzi!

ryanahamilton reviewed (changes requested):
Appears to be some related test failures:
```bash
 =================================== FAILURES ===================================
  _________________ TestExtraLinks.test_extra_link_in_gantt_view _________________
  
  self = <tests.www.test_views.TestExtraLinks testMethod=test_extra_link_in_gantt_view>
  
      def test_extra_link_in_gantt_view(self):
          exec_date = dates.days_ago(2)
          start_date = datetime(2020, 4, 10, 2, 0, 0)
          end_date = exec_date + timedelta(seconds=30)
      
          with create_session() as session:
              for task in self.dag.tasks:
                  ti = TaskInstance(task=task, execution_date=exec_date, state="success")
                  ti.start_date = start_date
                  ti.end_date = end_date
                  session.add(ti)
      
          url = f'gantt?dag_id={self.dag.dag_id}&execution_date={exec_date}'
          resp = self.client.get(url, follow_redirects=True)
      
  >       self.check_content_in_response('"extraLinks":', resp)
  
  tests/www/test_views.py:2940: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  self = <tests.www.test_views.TestExtraLinks testMethod=test_extra_link_in_gantt_view>
  text = '"extraLinks":', resp = <Response 46988 bytes [200 OK]>, resp_code = 200
  
      def check_content_in_response(self, text, resp, resp_code=200):
          resp_html = resp.data.decode('utf-8')
          assert resp_code == resp.status_code
          if isinstance(text, list):
              for line in text:
                  assert line in resp_html
          else:
  >           assert text in resp_html
  E           AssertionError: assert '"extraLinks":' in "\n\n\n\n\n\n\n\n\n\n    \n\n\n<!DOCTYPE html>\n<html>\n  <head>\n    <title>dag - Gantt - Airflow</title>\n\n    \n  ...   $input.addClass('switch-input--error');\n        }, 500 );\n      });\n    });\n  </script>\n\n\n  </body>\n</html>"
 
```

bbovenzi committed:
fix extra_links test failure

bbovenzi commented on airflow/www/templates/airflow/gantt.html:
```diff
@@ -76,20 +76,11 @@
-      });
-    });
+    // Loading data via <meta> wasn't loading extra_links in a task
+    const data = JSON.parse('{{ data|tojson }}');
```
@ryanahamilton Let me know if this is ok.

kaxil committed:
Fix broken docs build on Master (#14496)

Currently docs build are broken, example: https://github.com/apache/airflow/runs/1991279427

because of the following error:

```
  sphinx.errors.SphinxWarning: failed to reach any of the inventories with the following issues:
  intersphinx inventory '/opt/airflow/docs/_inventory_cache/apache-airflow-providers-tableau/objects.inv' not fetchable due to <class 'FileNotFoundError'>: [Errno 2] No such file or directory: '/opt/airflow/docs/_inventory_cache/apache-airflow-providers-tableau/objects.inv'
```

This PR fixes it and takes care of the case when inventory is not present, which happens when. a new provider is added. Once the docs are built the objects.env file is uploaded successfully.

kaxil committed:
Update docs about tableau and salesforce provider (#14495)

Since #14030 tableau is no longer deprecated, we can remove this doc entry

VBhojawala committed:
Corrects order of argument in docstring in GCSHook.download method (#14497)

piffall committed:
Adding support to put extra arguments for Glue Job. (#14027)

mik-laj committed:
Allow your own Docker production image to be verified by bash script (#14224)

Co-authored-by: Kamil Bregula <kamilbregula@Kamils-MacBook-Pro.local>

mik-laj committed:
Add docs about supported logging levels (#14507)

* Add docs about supported logging levels

* fixup! Add docs about supported logging levels

Co-authored-by: Kamil Bregula <kamilbregula@Kamils-MacBook-Pro.local>

kaxil committed:
Replace Stale Bot with Stale Github Action (#14494)

Looks like Stable Bot is not working anymore for us, so let's replace it with Github Action that runs everyday at 00:00

jbampton committed:
Fix spellings (#14483)

vemikhaylov committed:
Fix breeze redirect on macOS (#14506)

potiuk committed:
Prepare to release the next wave of providers: (#14487)

* Prepare to release the next wave of providers:

Regular providers:

* amazon          - 1.2.0
* apache.beam     - 1.0.1
* apache.druid    - 1.1.0
* apache.hive     - 1.0.2
* apache.spark    - 1.0.2
* cncf.kubernetes - 1.0.2
* dingding        - 1.0.2
* docker          - 1.0.2
* elasticsearch   - 1.0.2
* exasol          - 1.1.0
* google          - 2.1.0
* http            - 1.1.1
* jenkins         - 1.1.0
* microsoft.azure - 1.2.0
* mysql           - 1.0.2
* neo4j           - 1.0.1
* openfaas        - 1.1.1
* papermill       - 1.0.2
* presto          - 1.0.2
* qubole          - 1.0.2
* salesforce      - 2.0.0
* sendgrid        - 1.0.2
* sftp            - 1.1.1
* slack           - 3.0.0
* snowflake       - 1.1.1
* sqlite          - 1.0.2
* ssh             - 1.2.0
* tableau         - 1.0.0
* telegram        - 1.0.2

Backport providers (all in version 2021.3.3):

amazon apache.beam apache.druid apache.hive apache.spark cncf.kubernetes
dingding docker elasticsearch exasol google http jenkins microsoft.azure
mysql neo4j openfaas papermill presto qubole salesforce sendgrid sftp
slack snowflake sqlite ssh tableau telegram

* fixup! Prepare to release the next wave of providers:

potiuk committed:
Removes the step to upload artifact with documentation (#14510)

It takes insane 9:30 on the self-hosted runner and likely incurs
a lot of cost on pushing the docs (and it's value is limited)

mik-laj committed:
Update hadolint from v1.18.0 to v1.22.1 (#14509)

* Update hadolint from v1.22.1 to v1.18.0

* fixup! Update hadolint from v1.22.1 to v1.18.0

* fixup! fixup! Update hadolint from v1.22.1 to v1.18.0

Co-authored-by: Kamil Breguła <kamilbregula@apache.org>

mik-laj committed:
Production image can be run as root (#14226)

* Production image can be run as root

* fixup! Production image can be run as root

* fixup! fixup! Production image can be run as root

Co-authored-by: Kamil Bregula <kamilbregula@Kamils-MacBook-Pro.local>
Co-authored-by: Kamil Breguła <kamilbregula@apache.org>

mik-laj committed:
Enable LDAP auth in docker-compose.yaml (#14516)

Co-authored-by: Kamil Breguła <kamilbregula@apache.org>

jbampton committed:
chore: fix case of GitHub (#14525)

mik-laj committed:
Add health-check for celery worker (#14522)

Co-authored-by: Kamil Breguła <kamilbregula@apache.org>

jbampton committed:
Change HTTP to HTTPS in links (#14527)

turbaszek committed:
Make airflow info to work with pipes (#14528)

After this change output from AirflowConsole can be piped in bash
without loosing some information thanks to fixed width of output.

Closes: #14518


Co-authored-by: Kamil Breguła <mik-laj@users.noreply.github.com>

jbampton committed:
Add pre-commit check to sort and remove duplicates from the spelling wordlist (#13170)

potiuk committed:
Fix asset recompilation message (#14532)

The asset recompilation message did not work well in case of the
tests - where we did not mount local sources. It was always
showign the instructions to recompile the assets.

This is now fixed and the "OK" message is green.

mik-laj committed:
Add CLI check for scheduler (#14519)

Co-authored-by: Kamil Breguła <kamilbregula@apache.org>

ferruzzi committed:
Implemented S3 Bucket Tagging (#14402)

mik-laj committed:
Use airflow db check command in entrypoint_prod.sh (#14530)

* Use airflow db check in entrypoint_prod.sh

* fixup! Use airflow db check in entrypoint_prod.sh

Co-authored-by: Kamil Breguła <kamilbregula@apache.org>

radim94 committed:
Add Google leveldb hook and operator (#13109) (#14105)

* Add Google leveldb hook (#13109)

* Add write_batch, options for DB creation(comparator and other) plus fixes (#13109)

* Fix some static checks, add docs (#13109)

* Apply suggestions from code review

Co-authored-by: RosterIn <48057736+RosterIn@users.noreply.github.com>

* Fix some otger static checks and docs (#13109)

* Fix tests and some build-docs checks (#13109)

* Apply suggestions from code review

Co-authored-by: RosterIn <48057736+RosterIn@users.noreply.github.com>

* Fix build-docs checks (#13109)

* Fix build-docs checks (#13109)

* fixup! Fix build-docs checks (#13109)

* Rewrite example dag as in google package (#13109)

* Add extra options in operator, fix docstrings (#13109)

* Update airflow/providers/google/leveldb/operators/leveldb.py

Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

* Add system testing and docstrings (#13109)

* Fix comparator place in spelling wordlist(#13109)

Co-authored-by: RosterIn <48057736+RosterIn@users.noreply.github.com>
Co-authored-by: Kamil Bregula <kamilbregula@Kamils-MacBook-Pro.local>
Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

mik-laj committed:
Add docs about Celery monitoring (#14533)

Part of: #11161

To have a full description of the monitoring of all Airflow components, I add information about HTTP and CLI checks for Celery. Thanks to this, we will not have to search for information one by one, but everything will be in one place.

Documentation for Celery does not describe the inspect ping command, but hopefully, this will be added soon.

Commit:
BugFix: TypeError in monitor_pod (#14513)

If the log read is interrupted before any logs are produced then
`last_log_time` will not be set and the line
`delta = pendulum.now() - last_log_time` will fail with
```
TypeError: unsupported operand type(s) for -: 'DateTime' and 'NoneType'
```

This commit fix this issue by only updating `read_logs_since_sec` if
`last_log_time` has been set.

Abhik1998 committed:
Resolve issue related to HiveCliHook kill (#14542)

Closes: #13629
Fixed issue in Hive.py file

shizidushu committed:
WinRM Operator: Fix stout decoding issue & add option to run command (#13153)

Fix #13132 and add the ability to run command as powershell script.



Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

bbovenzi committed:
Merge branch 'master' into gantt-js

bbovenzi committed:
spelling

bbovenzi committed:
fix conflict

bbovenzi committed:
Merge branch 'gantt-js' of https://github.com/astronomer/airflow into gantt-js
